tests/contracts/equal.clar

```diff
@@ -53,3 +53,147 @@
+)
+
+(define-public (call-principal-equal)
+    (ok (is-eq 'ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM.foo 'ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM.foo))
+)
+
+(define-public (call-principal-unequal)
+    (ok (is-eq 'ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM.foo 'ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM.bar))
+)
+
+(define-public (call-principal-unequal-2)
+    (ok (is-eq 'ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM.foo 'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR.foo))
+)
+
+(define-public (call-optional-equal)
+    (ok (is-eq (some 1) (some 1)))
+)
+
+(define-public (call-optional-unequal)
+    (ok (is-eq (some 0x01) (some 0x02)))
+)
+
+(define-public (call-response-ok-equal)
+    (ok (is-eq (ok 0) (ok 0)))
+)
+
+(define-public (call-response-err-equal)
+    (ok (is-eq (err 0x010203) (err 0x010203)))
+)
+
+(define-public (call-response-ok-err-unequal)
+    (ok (is-eq (ok 42) (err "forty-two")))
+)
+
+(define-public (call-response-ok-unequal)
+    (ok (is-eq (ok u5) (ok u55)))
+)
+
+(define-public (call-response-err-unequal)
+    (ok (is-eq (err 0x123456) (err 0xabcdef)))
+)
+
+(define-public (call-one-tuple-equal)
+    (ok (is-eq {name: "James Bond"} {name: "James Bond"}))
+)
+
+(define-public (call-one-tuple-unequal)
+    (ok (is-eq {name: "James Bond"} {name: "Austin Powers"}))
+)
+
+(define-public (call-tuple-equal)
+    (ok (is-eq {name: "James Bond", alias: "007"} {name: "James Bond", alias: "007"}))
+)
+
+(define-public (call-tuple-unequal)
+    (ok (is-eq {name: "James Bond", alias: "007"} {name: "James Bond", alias: "008"}))
+)
+
+(define-public (call-big-tuple-equal)
+    (ok (is-eq {name: "Hubert Bonisseur de la Bath", alias: "OSS 117", creation: 1949} {name: "Hubert Bonisseur de la Bath", alias: "OSS 117", creation: 1949}))
+)
+
+(define-public (call-big-tuple-slightly-unequal)
+    (ok (is-eq {name: "Hubert Bonisseur de La Bath", alias: "OSS 117", creation: 1949} {name: "Hubert Bonisseur de la Bath", alias: "OSS 117", creation: 1949}))
+)
+
+(define-public (call-big-tuple-unequal)
+    (ok (is-eq {name: "Hubert Bonisseur de la Bath", alias: "OSS 117", creation: 1949} {name: "James Bond", alias: "007", creation: 1953}))
+)
+
+(define-public (call-tuple-recursive-equal)
+    (ok (is-eq {name: "James Bond", creator: {name: "Ian Fleming", date: (some 1953)}} {name: "James Bond", creator: {name: "Ian Fleming", date: (some 1953)}}))
+)
+
+(define-public (call-tuple-recursive-unequal)
+    (ok (is-eq {name: "James Bond", creator: {name: "Ian Fleming", date: (some 1953)}} {name: "James Bond", creator: {name: "Ian Fleming", date: none}}))
+)
+
+(define-public (call-list-int-equal)
+    (ok (is-eq (list 1 2 3 4 5) (list 1 2 3 4 5)))
+)
+
+(define-public (call-list-str-equal)
+    (ok (is-eq (list "hello" "world" "goodbye" "mars") (list "hello" "world" "goodbye" "mars")))
+)
+
+(define-public (call-list-unequal-length)
+    (ok (is-eq (list 1 2 3 4 5) (list 1 2 3 4)))
+)
```

Comment:
Let's also include one with these same arguments but in the opposite order. This would be helpful if the algorithm ever changes such that the length isn't checked separately first.

Reply:
added in 93aa27cdcbd93d5a5b3ccbbd7c26aacc5f47f195